keps/sig-node/3619-supplemental-groups-policy/kep.yaml

```diff
@@ -0,0 +1,43 @@
+title: "Fine grained SupplementalGroups control"
+kep-number: 3619
+authors:
+  - "@everpeace"
+owning-sig: sig-xyz
+participating-sigs:
+  - sig-node
+status: provisional
+creation-date: 2022-10-14
+reviewers:
+  - "@thockin"
+  - "@mrunalp"
+approvers:
+  - TBD
```

Comment:
Need a volunteer for approver - @mrunalp ? @SergeyKanzhelev ?

Reply:
I don't think I can approve the KEP. But I would be interested to be a reviewer. It is important to think thru customers transition - how they detect the potentially problematic images and decide they are ready to make the bug fixed

Reply:
> But I would be interested to be a reviewer.

@SergeyKanzhelev I added you as an reviewer in kep.yaml: 7939edb72dc89efd529238dce70891eeaae3b006

> how they detect the potentially problematic images and decide they are ready to make the bug fixed

Probably, they have to check manually group information by invoking `id` command in their pods...

Reply:
Good question Sergey.

Would it be reasonable to add a few fields to CRI (`message Container` via `ListContainersResponse`) which indicate:

1) The running UID (in and out of userns)
2) The runnign GID (in and out of userns)
3) The supplemental GIDs (in and out of userns)

?

Then kubelet has information which we can put back into PodStatus, maybe even a Condition like "SupplementalGroupsFromImage".

Also maybe a metric for number of pods which have supplemental groups from image.

Reply:
You can add me as approver. Thanks!

Reply:
If we go with the fix in Container Runtime, perhaps we do detection from NPD instead of changing kubelet at all. Can this information be extracted from the host without changing CRI?

Reply:
> You can add me as approver. Thanks!

@mrunalp Thanks! added you as an approver in 342a69873ab032c80c43858f094b3905f6ce6559

> Can this information be extracted from the host without changing CRI?

If we could resolve the container's OCI [filesystem bundle](https://github.com/opencontainers/runtime-spec/blob/main/bundle.md) path in some way, we should be able to find [`config.json`](https://github.com/opencontainers/runtime-spec/blob/main/config.md) in the bundle and can inspect [`User`](https://github.com/opencontainers/runtime-spec/blob/main/config.md#user). We can detect gids not in Pod spec (i.e. gids from the image).

But, we should find bundle paths in some way which is independent to OCI runtimes (both high level(e.g. containerd) and low level(e.g. runc)).

Reply:
I see. I think this consideration as written needs to be placed in "alternatives" section. Maybe with more details why it is less reliable. Especially if we are talking about extending CRI for this reporting.